Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,231 @@
+// Created by Dan Federman on 2/2/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import SwiftSyntax
+
+public enum TypeDescription: Codable, Equatable, CustomStringConvertible {
+  case simple(name: String)
+  indirect case member(name: String, baseType: TypeDescription)
+  indirect case composition([TypeDescription])
+  indirect case optional(TypeDescription)
+  indirect case implicitlyUnwrapped(TypeDescription)
+  indirect case array(TypeDescription)
+  indirect case dictionary(key: TypeDescription, value: TypeDescription)
+  indirect case tuple([TypeDescription])
+  case unknown(text: String)
```

Comment:
Note that I'm still missing cases for the following types:
```
 ClassRestrictionTypeSyntax
 SomeTypeSyntax
 MetatypeTypeSyntax
 FunctionTypeSyntax
 AttributedTypeSyntax
 UnknownTypeSyntax
```

This PR is big enough. We can add these types later as needed.

Reply:
What do you think about adding code comment to indicate these intentional omissions?

Reply:
I love it.

dd148f5f47478e951c361e750081533bb9e6b81d